src/renderer/store/modules/sponsorblock.js

```diff
@@ -0,0 +1,38 @@
+import $ from 'jquery'
+import forge from 'node-forge'
+
+const state = {}
+const getters = {}
+
+const actions = {
+  sponsorBlockSkipSegments ({ rootState }, { videoId, categories }) {
+    return new Promise((resolve, reject) => {
+      const messageDigestSha256 = forge.md.sha256.create()
+      messageDigestSha256.update(videoId)
+      const videoIdHashPrefix = messageDigestSha256.digest().toHex().substring(0, 4)
+      const requestUrl = `${rootState.settings.sponsorBlockUrl}skipSegments/${videoIdHashPrefix}?categories=${JSON.stringify(categories)}`
```

Comment:
This logic assumes that the URL stored in settings will always end in a forward slash. If I remove the forward slash in my settings, then this logic will break. Let's sanitize the input to never have a forward slash and then put it here. Check out the Invidious instance logic for assistance with this.

On another note, I see that the default instance includes `/api/`. What I'd like to know is if this is specific to the main Sponsorblock site or if all selfhosted Sponsorblock sites keep everything in the `/api/` endpoint? If the latter, let's add `/api/` to the hard coded section instead of making the user remember to include that portion.

Reply:
Selfhosted versions will keep the `/api/` in the url.

Reply:
I remove trailing slash and /api suffix from sponsorBlockUrl in https://github.com/FreeTubeApp/FreeTube/pull/1130/commits/60a01a7c07186217ec235d38ba3bcb98f239933d